benchexec/tools/depthk.py

```diff
@@ -0,0 +1,159 @@
+    """
+
+    REQUIRED_PATHS = [
+        'depthk.py',
+        'depthk-wrapper.sh',
+        'esbmc',
+        '__init__.py',
+        'modules',
+        'tokenizer',
+        ]
+
+    def executable(self):
+
+        # Relative path to depthk wrapper
+
+#        return Util.find_executable('../../esbmc+depthk/depthk-wrapper.sh')
+
+        return Util.find_executable('depthk-wrapper.sh'
+                                    )
+
+    def working_directory(self, executable):
+        executableDir = os.path.dirname(executable)
+        return executableDir
+
+    def environment(self, executable):
+        return {'additionalEnv': {'PATH': ':.'}}
+
+    def version(self, executable):
+        workingDir = self.working_directory(executable)
+
+        version = subprocess.Popen([workingDir + '/depthk.py',
+                                   '--version'],
+                                   stdout=subprocess.PIPE,
+                                   stderr=subprocess.STDOUT).stdout.readline().decode()
+        return version
+
+    def name(self):
+        return 'DepthK'
+
+    def cmdline(
+        self,
+        executable,
+        options,
+        tasks,
+        propertyfile,
+        rlimits,
+        ):
+
+        assert len(tasks) == 1, 'only one sourcefile supported'
+        assert propertyfile, 'property file required'
+        sourcefile = tasks[0]
+        return [executable] + options + ['-c', propertyfile, sourcefile]
+
+    def determine_result(
+        self,
+        returncode,
+        returnsignal,
+        output,
+        isTimeout,
+        ):
+
+        if len(output) <= 0:
+            return
+
+        output = output[-1].strip()
+        status = ''
+
+        if 'TRUE' in output:
+            status = result.RESULT_TRUE_PROP
+        elif 'no-overflow' in output:
+            status = result.RESULT_FALSE_OVERFLOW
+        elif 'valid-deref' in output:
+            status = result.RESULT_FALSE_DEREF
+        elif 'valid-memtrack' in output:
+            status = result.RESULT_FALSE_MEMTRACK
+        elif 'FALSE' in output:
+            status = result.RESULT_FALSE_REACH
+        else:
+            status = result.RESULT_UNKNOWN
+
+        # else:
+        #    status = result.RESULT_ERROR
+
+        return status
+
+    def get_value_from_output(self, lines, identifier):
+
+        # search for the text in output and get its value,
+        # stop after the first line, that contains the searched text
+
+        for line in lines:
+            if identifier == 'k' and line.startswith('Bound k:'):
+                matchbound = re.search(r'Bound k:(.*)', line)
+                return matchbound.group(1).strip()
+            if identifier == 'Step' and line.startswith('Solution by:'):
+                matchstep = re.search(r'Solution by:(.*)', line)
+                return matchstep.group(1).strip()
+
+        return '-'
+
+    def addColumnValues(self, output, columns):
```

Comment:
The method `addColumnValues` was only used a long time ago. It was replaced by `get_value_from_output`, so please remove it.